test/e2e/custom_wait_rules_test.go

```diff
@@ -119,3 +120,144 @@ status:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                cronSpec:
+                  type: string
+                image:
+                  type: string
+                replicas:
+                  type: integer
+            status:
+              type: object
+              properties:
+                currentState:
+                  type: string
+  scope: Namespaced
+  names:
+    plural: crontabs
+    singular: crontab
+    kind: CronTab
+---
+apiVersion: "stable.example.com/v1"
+kind: CronTab
+metadata:
+  name: my-new-cron-object-1
+  annotations:
+    kapp.k14s.io/change-group: "cr"
+spec:
+  cronSpec: "* * * * */5"
+  image: my-awesome-cron-image
+status:
+  currentState: Progressing
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test-cm
+  annotations:
+    kapp.k14s.io/change-rule: "upsert after upserting cr"`
+
+	name := "test-custom-wait-rule-contract-v1"
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+
+	cleanUp()
+	defer cleanUp()
+
+	go func() {
+		for {
+			time.Sleep(1 * time.Second)
+			exists, err := ClusterResourceExists("ConfigMap", "test-cm", kubectl)
+			require.NoError(t, err, "Expected error to not have occurred")
+			if exists {
+				break
+			}
+		}
+		patch := `[{ "op": "replace", "path": "/status/currentState", "value": "Running"}]`
+		PatchClusterResource("CronTab", "my-new-cron-object-1", env.Namespace, patch, kubectl)
+	}()
+
+	logger.Section("deploy resource with current state as progressing", func() {
+		out, _ := kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, RunOpts{
+			StdinReader: strings.NewReader(yaml)})
+
+		out = sortActionsInResult(strings.TrimSpace(replaceSpaces(replaceTarget(replaceTs(out)))))
+
+		expectedOutput := strings.TrimSpace(replaceSpaces(`Changes
+
+Namespace  Name                         Kind                      Age  Op      Op st.  Wait to    Rs  Ri  $
+(cluster)  crontabs.stable.example.com  CustomResourceDefinition  -    create  -       reconcile  -   -  $
+kapp-test  my-new-cron-object-1         CronTab                   -    create  -       reconcile  -   -  $
+^          test-cm                      ConfigMap                 -    create  -       reconcile  -   -  $
+
+Op:      3 create, 0 delete, 0 update, 0 noop, 0 exists
+Wait to: 3 reconcile, 0 delete, 0 noop
+
+<replaced>: ---- applying 1 changes [0/3 done] ----
+<replaced>: create customresourcedefinition/crontabs.stable.example.com (apiextensions.k8s.io/v1) cluster
+<replaced>: ---- waiting on 1 changes [0/3 done] ----
+<replaced>: ok: reconcile customresourcedefinition/crontabs.stable.example.com (apiextensions.k8s.io/v1) cluster
+<replaced>: ---- applying 1 changes [1/3 done] ----
+<replaced>: create crontab/my-new-cron-object-1 (stable.example.com/v1) namespace: kapp-test
+<replaced>: ---- waiting on 1 changes [1/3 done] ----
+<replaced>:  ^ Unblock blocked changes
```

Comment:
hmm, may be we should have some consistency between ytt and non-ytt unblock messaging. e.g.

`Allowing blocked changes to proceed: <ytt user message>`
`Allowing blocked changes to proceed: Encountered condition %s == %s: reason`

there should also be a line describing resource above the `^`. but there isnt in this output.

Reply:
> Apparently, the order of lines is messed up (and it's consistent, so I am guessing it's happening in one of the replace functions).

`sortActionsInResult` was causing this, not sure why I added it in the first place (maybe I just copy pasted it from another test). I have removed it and the order if fixed now.